src/wp-includes/media.php

```diff
@@ -1854,7 +1854,14 @@ function wp_filter_content_tags( $content, $context = null ) {
 
 			// Use alternate mime types when specified and available.
 			if ( $attachment_id > 0 && _wp_in_front_end_context() ) {
+				$original_image = $filtered_image;
 				$filtered_image = wp_image_use_alternate_mime_types( $filtered_image, $context, $attachment_id );
+				if (
+					! has_action( 'wp_footer', 'wp_wepb_fallback' ) &&
+					$original_image !== $filtered_image
+				) {
+					add_action( 'wp_footer', 'wp_wepb_fallback' );
+				}
```

Comment:
There are some critical conditions missing here which are present in the plugin. For now, we should only print this script when any JPEG image is replaced with a WebP, since the script only helps with that.

Potentially it would be better for the core implementation to handle this inside the `wp_image_use_alternate_mime_types()` function as well. But for better parity with core behavior, I would suggest the following:
* In `wp_image_use_alternate_mime_types()`, never add the action to `wp_footer`, but only set a (private) global flag like `$_wp_image_mime_fallback_should_load = true`.
* Add the action unconditionally inside of `default-filters.php`.
* Inside of the action callback function, check for the global and bail early if it's not set or `false`.

Reply:
Good point. 

> In wp_image_use_alternate_mime_types(), never add the action to wp_footer, but only set a (private) global flag like $_wp_image_mime_fallback_should_load = true.

Note this should only happen towards the end of the function, if either the full sized or one of the srcset sub sized urls was swapped (L1969, L1997). _We shouldn't set this if the image is returned unchanged_, since it is only useful if we swapped image urls here.

Reply:
Thank you for the clear instructions for the update.